src/components/binding/binding-panel.js

```diff
@@ -213,13 +187,51 @@ genotet.BindingPanel.prototype.updateTracks = function() {
       this.container.find('#genes #track-' + index).remove();
     }
   }
-  this.data.tracks.forEach(function(track, index) {
-    var ui = this.container.find('#genes #track-' + index);
-    if (!ui.length) {
-      this.addTrack_();
-    }
-    this.selectGenes_[index].val(track.fileName).trigger('change');
-  }, this);
+
+  // Load binding list and update panel tracks.
+  var params = {
+    type: 'list-binding'
+  };
+  $.get(genotet.data.serverURL, params, function(data) {
+    genotet.data.bindingFiles = [];
+    data.forEach(function(dataInfo) {
+      genotet.data.bindingFiles.push(dataInfo);
+    });
+    var fileNames = genotet.data.bindingFiles.map(function(dataInfo) {
+      return {
+        id: dataInfo.fileName,
+        text: dataInfo.gene + ' (' + dataInfo.fileName + ')'
+      };
+    });
+
+    this.data.tracks.forEach(function(track, index) {
```

Comment:
Try to avoid putting chunks of code in Ajax callback. Write a separate function for each processing step.